### DISCUSSION TITLE
Mac Homebrew installer

### DISCUSSION THREAD

**Discussion by kalenpeterson** [openscopeproject/TrguiNG#150] [Ideas]
Hey!

Just found this as a replacement for transmission-remote-gui on Mac. Works great. Finally!

I noticed there does not seem to be a homebrew installer submitted for this yet, and MacOS installs are manual. Has anyone considered or started working on that yet? If not I would be happy to build it and submit a PR to Homebew.

https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request

Thanks!

**Comment by qu1ck** [comment]
How would updating the recipe (cask?) to newer version work? You would need to open new PR against homebrew repo every time? Is there a way to automate it without involvement of third parties?

**Reply by kalenpeterson** [reply]
Great questions. I have not written a cask before, so I don't know all the details yet. If no one else is working on this I will look into it and propose a solution.

I would want it automated. I don't want to manually make PRs for every release. Ideally, it would integrate into this project's release pipeline. My primary reason for wanting this is to have automatic updates via homebew. I do not want to have to check for updates and install them manually.

**Reply by qu1ck** [reply]
No one else is working on this. Ideally it will integrate into release pipeline (this is how arch aur works), second best option is to have me doing some manual update without involving anyone else (this is how flathub update works).

Tauri also supports update checks and self updates in the framework, I had it disabled while the app was in beta and because it had some issues. Maybe it's time to look into it again.

**Reply by kalenpeterson** [reply]
So, I did some digging on this. Here is what would need to be done.

1. Develop the cask recipe based on current version
2. Manually submit a PR to homebrew/cask for review
3. Add a step to the release pipeline to update the version

After the first PR is merged, they offer a tool to automate updating the version with one command. Under the covers, this submits a new PR to Homebrew with the version bump, but it can just be added to the release pipeline. It looks like these bump PRs get merged pretty quickly.

Also, there is an auto_updates option that can be enabled in the cask recipe. If you chose to turn on self updates in Tauri, enabling this will tell homebrew not to continue updating the app itself, just do the initial install.

If this is of interest, I'm happy to work on the initial cask/PR, and figure out what needs to be added to the pipeline.

References
- https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request
- https://openrefine.org/docs/technical-reference/homebrew-cask-process
- https://docs.brew.sh/Cask-Cookbook

**Reply by qu1ck** [reply]
Thanks. Something similar was done for arch aur. A contributor set up the initial build script which they pushed to AUR repo. Then they added me as contributor and I setup a github action that automatically updates it using my key.

This is similar homebrew action to push a formula update PR
https://github.com/Homebrew/actions/tree/master/bump-packages 

If you can setup the recipe, add me as maintainer if there is any sort of access list and we can add the update step here 
https://github.com/openscopeproject/TrguiNG/blob/master/.github/workflows/build.yml
Only downside seems to be that it's still a PR so someone at the minimum needs to merge it so it's not fully automated but it's not in our control.

**Reply by DuncanHills** [reply]
@kalenpeterson do you still plan on working on this?
